### PR TITLE
wsManager: Fix IsRunning true before flushed

### PR DESCRIPTION
**Pull request by gbjk**
#### wsManager: Fix IsRunning true before flushed

Consumers must be able to tell when it's safe to start new
 subscriptions. Before this fix any new subscriptions would get unsubbed
 as part of the flush during `websocketRoutine`.
                                                                                                                                                                                                 
This fix also significantly simplifies the state management, removing
 atomics in favour of using the existing mutex as per
 the [atomics docs](https://pkg.go.dev/sync/atomic#pkg-overview).

This fix does not require this simplification but it's much clearer with
 it. Specifically in Stop() where we can't do an atomic swap because we

#### wsManager: Remove duplicate shutdown chan make
    
The shutdown channel was already in setup.
Consumers could reasonably expect it to not be replaced in between the
two, and it's not really part of Start to assign it.

####  Engine: Expose WebsocketRoutineManager

Without exposing the manager users don't know when it's ready to use.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How has this been tested

- [x] go test ./... -race

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by gbjk** [comment]
Hold off on reviewing this.
I see a cleaner solution which retains most of the atomics code.
Will update shortly

**Comment by gbjk** [comment]
Okay. Apologies for the brain spam here. All sorted now with the least invasive fix for the problem.
Ready for review.

**Review by gloriousCode — COMMENTED** [review]
I like your updated PR 🎉

File: engine/engine.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ type Engine struct {
 	OrderManager            *OrderManager
 	portfolioManager        *portfolioManager
 	gctScriptManager        *gctscript.GctScriptManager
-	websocketRoutineManager *websocketRoutineManager
+	WebsocketRoutineManager *websocketRoutineManager
```
[the remark this answers is not in the document]
Comment (gbjk) [reply engine/engine.go:47]: Nope. Just that, status.

Could wrap wsRoutineManagerReady as a method on the engine I suppose.

This architecture is a little unusual to me, in that these things are part of the engine package.
Normally they'd be their own package and probably the engine would have them as an interface type.
But this seems to be working so 🤷

[the remark this answers is not in the document]
Comment (gloriousCode) [reply engine/engine.go:47]: I've been stuck thinking on exporting an unexported property. I think if you just export WebsocketRoutineManager its all good. There's not much utility for it outside of `engine` package for it anyway, and it improves your workflow

[the remark this answers is not in the document]
Comment (gbjk) [reply engine/engine.go:47]: I'm not sure I see your concern.
However I also don't have a horse in the race so happy to change it.

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks for making those changes

**Review by thrasher- — APPROVED** [review]

**Pull request merged**